File: PracticasProgramacion/Practica_16/python/main.py
"""
Practica_16 - Calculadora recursiva y fractales.
"""

from calculadora_recursiva import CalculadoraRecursiva
from fractales import Fractales

if __name__ == "__main__":
    calc = CalculadoraRecursiva()
    print(f"4*5 (sumas recursivas): {calc.multiplicar(4.0, 5.0)}")
    print(f"2^4 (mults recursivas): {calc.potencia(2.0, 4)}")
    print(f"15/3 (restas recursivas): {calc.dividir(15.0, 3.0)}")

    frac = Fractales()
    frac.triangulo_sierpinski(3)
    print("\nPolvo de Cantor nivel 2:")
    frac.polvo_cantor(2, 0)

File: PracticasProgramacion/Practica_16/cpp/main.cpp
/**
 * Practica_16 - Calculadora recursiva + Fractales (Sierpinski, Cantor)
 */

#include "CalculadoraRecursiva.h"
#include "Fractales.h"
#include <iostream>

int main() {
    CalculadoraRecursiva calc;
    std::cout << "4*5 (sumas recursivas): " << calc.multiplicar(4.0, 5.0) << std::endl;
    std::cout << "2^4 (mult recursivas): " << calc.potencia(2.0, 4) << std::endl;
    std::cout << "15/3 (restas recursivas): " << calc.dividir(15.0, 3.0) << std::endl;

    Fractales frac;
    frac.trianguloSierpinski(3);
    std::cout << "\nPolvo de Cantor nivel 2:\n";
    frac.polvoCantor(2, 0);

    return 0;
}
